tests/drivers/adc/adc_api/src/test_adc.c

```diff
@@ -377,7 +392,7 @@ static int test_task_asynchronous_call(void)
 	const struct adc_sequence_options options = {
 		.extra_samplings = 4,
 		/* Start consecutive samplings as fast as possible. */
-		.interval_us     = 0,
+		.interval_us     = REPEATED_SAMPLING_INTERVAL,
```

Comment:
The idea of this test case is to perform samplings without any interval between them, to cover this API requirement: https://github.com/zephyrproject-rtos/zephyr/blob/2ec5ec2153f91216482b1993f75eebc1ea4418bb/include/drivers/adc.h#L236-L237
It will look a bit weird after the above change.

Reply:
From my point of view, this functions aims to test asynchronous call.
So even if there is an interval, it doesn't matter, isn't it ?
Other alternative is to say that measuring adc floating input gives no guarantee of non zero value (from hardware point of view), and thus test can't be passed. And I would like to avoid exclude STM32L0 boards from this test.

Reply:
The case of no interval should be tested on STM32L0 as well, to ensure it is handled properly on this SoC.
I'd rather modify the test in a different way. The `check_samples()` function checks if only the required entries in the sampling buffer were filled by the driver. Instead of 0, which was apparently not a good choice for marking entries as empty, we could use some other value not supposed to be returned by drivers. Perhaps something like `SHRT_MIN` could work, as almost all drivers use a resolution less than 16 bits in this test. And something like this could be used:
```
#if !defined(INVALID_ADC_VALUE)
#define INVALID_ADC_VALUE SHRT_MIN
#endif
```
to give the possibility to use a specific value for a particular driver if needed.
What you think? I'll prepare a PR with such modification.

Reply:
https://github.com/zephyrproject-rtos/zephyr/pull/28958

Reply:
Hi @anangl,
It sounds good to me.
I test it on nucleo_l073rz, nucleo_l152re, nucleo_wb55rg
It is all passed.